lib/misc/api/api_user.dart

```diff
@@ -42,3 +43,39 @@ Future<List<BaseItem>> fetchLocations() async {
 
   return location;
 }
+
+/// Check if the user has permission
+Future<bool> hasPermission({required String permission}) async {
+  final token = await api_token.getToken();
+  try {
+    final response = await http.get(
+      Uri.parse('${api.getApiBaseUrl()}/auth/ability/$permission'),
+      headers: <String, String>{
+        'Content-Type': 'application/json; charset=UTF-8',
+        'Accept': 'application/json',
+        'Authorization': 'Bearer $token',
+      },
+    );
+
+    final data = jsonDecode(response.body);
+    debugPrint(data.toString());
+    if (data['has_ability'] == true) {
+      return true;
+    }
+
+    if (response.statusCode != 200) {
+      if (kDebugMode) {
+        debugPrint('Request failed with status: ${response.statusCode}');
+        log('Request failed with status: ${response.statusCode}');
+      }
+    }
+  } catch (e) {
+    // Handle any exceptions that occur
+    if (kDebugMode) {
+      debugPrint('Error: $e');
+      log('Error: $e');
+    }
```

Comment:
Same as previous comment regarding returning future error. But also why log and debugprint?

Reply:
Because if it fails, it should just return false.